### PR TITLE
Auto merge backport PRs

### LINKED ISSUE #13666: Automatically merge backport PRs
**Description**

Backport PRs that are created by the backport-action could be merged automatically if they succeed the CI. At this stage the code has already been reviewed and if the CI fails there is no reason to not merge it.

We already have a similar step to merge dependabot PRs, which we can reuse by simply adding the backport-action user to it. It works by approving the PR using the `GITHUB_TOKEN` of the workflow and adding a `bors merge` comment to merge the PR.

### PR DESCRIPTION
**Pull request by remcowesterhoud**
## Description

Autoamtically merges backport PRs created by the backport-action user.

## Related issues

closes #13666 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [x] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by megglos — APPROVED** [review]
🚀  thanks for driving this!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -681,9 +682,15 @@ jobs:
...
-        if: steps.metadata.outputs.update-type == 'version-update:semver-patch' || steps.metadata.outputs.update-type == 'version-update:semver-minor'
+      - id: approve-and-merge-dependabot
+        name: Approve and merge dependabot PR
+        if: github.actor == 'dependabot[bot]' && (steps.metadata.outputs.update-type == 'version-update:semver-patch' || steps.metadata.outputs.update-type == 'version-update:semver-minor')
+        run: gh pr review ${{ github.event.pull_request.number }} --approve -b "bors merge"
+        env:
+          GITHUB_TOKEN: "${{secrets.GITHUB_TOKEN}}"
+      - id: approve-and-merge-backport
+        name: Approve and merge backport PR
+        if: github.actor != 'dependabot[bot]'
```
[the remark this answers is not in the document]
Comment (megglos) [reply .github/workflows/ci.yml:693]: yeah I would be for the more explicit approach right now, if we extend we can reiterate conditions

[the remark this answers is not in the document]
Comment (remcowesterhoud) [reply .github/workflows/ci.yml]: Fine with me, I've changed it

**Comment by remcowesterhoud** [comment]
bors merge

**Pull request merged**